code/modules/food_and_drink/snacks.dm

```diff
@@ -208,7 +208,36 @@
+	var/static/list/drinkwhitelist = list(/obj/item/reagent_containers/food/drinks/energyshake,/obj/item/reagent_containers/food/drinks/bottle/red,/obj/item/reagent_containers/food/drinks/bottle/blue,/obj/item/reagent_containers/food/drinks/bottle/pink,/obj/item/reagent_containers/food/drinks/bottle/lime,/obj/item/reagent_containers/food/drinks/bottle/spooky,/obj/item/reagent_containers/food/drinks/bottle/spooky2,/obj/item/reagent_containers/food/drinks/bottle/xmas,/obj/item/reagent_containers/food/drinks/bottle/bottledwater,/obj/item/reagent_containers/food/drinks/bottle/softsoft_pizza,/obj/item/reagent_containers/food/drinks/bottle/grones,/obj/item/reagent_containers/food/drinks/bottle/orange,/obj/item/reagent_containers/food/drinks/bottle/gingerale,/obj/item/reagent_containers/food/drinks/bottle/drowsy,/obj/item/reagent_containers/food/drinks/water,/obj/item/reagent_containers/food/drinks/tea,/obj/item/reagent_containers/food/drinks/coffee,/obj/item/reagent_containers/food/drinks/eggnog,/obj/item/reagent_containers/food/drinks/chickensoup,/obj/item/reagent_containers/food/drinks/weightloss_shake,/obj/item/reagent_containers/food/drinks/cola,/obj/item/reagent_containers/food/drinks/cola/random,/obj/item/reagent_containers/food/drinks/peach,/obj/item/reagent_containers/food/drinks/milk,/obj/item/reagent_containers/food/drinks/milk/rancid,/obj/item/reagent_containers/food/drinks/milk/soy,/obj/item/reagent_containers/food/drinks/covfefe,/obj/item/reagent_containers/food/drinks/bottle/beer,/obj/item/reagent_containers/food/drinks/bottle/fancy_beer,/obj/item/reagent_containers/food/drinks/bottle/wine,/obj/item/reagent_containers/food/drinks/bottle/hobo_wine,/obj/item/reagent_containers/food/drinks/bottle/champagne,/obj/item/reagent_containers/food/drinks/bottle/cider,/obj/item/reagent_containers/food/drinks/bottle/rum,/obj/item/reagent_containers/food/drinks/rum_spaced,/obj/item/reagent_containers/food/drinks/bottle/mead,/obj/item/reagent_containers/food/drinks/bottle/vintage,/obj/item/reagent_containers/food/drinks/bottle/vodka,/obj/item/reagent_containers/food/drinks/bottle/tequila,/obj/item/reagent_containers/food/drinks/bottle/gin,/obj/item/reagent_containers/food/drinks/bottle/ntbrew,/obj/item/reagent_containers/food/drinks/bottle/thegoodstuff,/obj/item/reagent_containers/food/drinks/curacao,)
+	proc/filter_food(food_type)
+		for (var/food_filter in src.blacklist)
+			if (ispath(food_type,food_filter))
+				return FALSE
+			.= TRUE
```

Comment:
I don't think you need to explicitly loop here, just do a `.Find`? Might be weird with paths as list items, but should be okay?

Reply:
I'm not aware of how to use .Find I'll try to look into it

Reply:
don't use .Find, use `in`

Reply:
Do what ZeWaka said, I know nothing.

Reply:
As ZeWaka says,  it is